File: nas-material-manager/flutter/lib/data/models/material_tags.dart
enum UsageTag {
  unused('unused', '未使用'),
  used('used', '使用过');

  final String rawValue;
  final String displayName;
  const UsageTag(this.rawValue, this.displayName);

  static UsageTag fromRawValue(String value) {
    return UsageTag.values.firstWhere(
      (e) => e.rawValue == value,
      orElse: () => UsageTag.unused,
    );
  }
}

enum ViralTag {
  notViral('not_viral', '未爆'),
  viral('viral', '爆款');

  final String rawValue;
  final String displayName;
  const ViralTag(this.rawValue, this.displayName);

  static ViralTag fromRawValue(String value) {
    return ViralTag.values.firstWhere(
      (e) => e.rawValue == value,
      orElse: () => ViralTag.notViral,
    );
  }
}

class MaterialTags {
  final UsageTag usage;
  final ViralTag viral;

  MaterialTags({
    required this.usage,
    required this.viral,
  });

  MaterialTags copyWith({
    UsageTag? usage,
    ViralTag? viral,
  }) {
    return MaterialTags(
      usage: usage ?? this.usage,
      viral: viral ?? this.viral,
    );
  }
}

File: nas-material-manager/ios/NASMaterialManager/NASMaterialManager/Domain/Models/MaterialTags.swift
import Foundation

enum UsageTag: String, Codable, CaseIterable {
    case unused = "unused"
    case used = "used"

    var displayName: String {
        switch self {
        case .unused: return "未使用"
        case .used: return "使用过"
        }
    }
}

enum ViralTag: String, Codable, CaseIterable {
    case notViral = "not_viral"
    case viral = "viral"

    var displayName: String {
        switch self {
        case .notViral: return "未爆"
        case .viral: return "爆款"
        }
    }
}

struct MaterialTags: Codable {
    var usage: UsageTag
    var viral: ViralTag
}
